modules/imaging_browser/php/viewsession.class.inc

```diff
@@ -55,17 +55,19 @@ class ViewSession extends \NDB_Form
         $candidate = \Candidate::singleton($candid);
 
         if ($candidate->getData('Entity_type') == 'Scanner') {
-            return $user->hasPermission('imaging_browser_phantom_allsites')
+            return ($user->hasPermission('imaging_browser_phantom_allsites')
                 || $user->hasCenterPermission(
                     'imaging_browser_phantom_ownsite',
                     \TimePoint::singleton($_REQUEST['sessionID'])->getCenterID()
-                );
+                ))
+                && ($user->hasProject($candidate->getProjectID()));
         } elseif ($candidate->getData('Entity_type') == 'Human') {
-            return $user->hasPermission('imaging_browser_view_allsites')
+            return ($user->hasPermission('imaging_browser_view_allsites')
                 || $user->hasCenterPermission(
                     'imaging_browser_view_site',
                     \TimePoint::singleton($_REQUEST['sessionID'])->getCenterID()
-                );
+                ))
+                && $user->hasProject($candidate->getProjectID());
```

Comment:
@cmadjar can you review? It looks like the code does what the description says but I'm not sure what the intended behaviour for the imaging browser is with the view all sites permission.

Reply:
This is a good point @driusan . I assumed all-sites did not apply to projects, but if all-sites means more along the lines of "can see everything", then this will need to be changed slightly, as well as #6640

Reply:
How is this handled for other modules? Are there project permissions implemented? If not, I would not implement add project permissions for this module in this release and wait until we properly deal with permissions.

The permissions of that module are already a spider web for that module with:
- view all sites
- view own sites
- view phantom all sites
- view phantom own sites

If we go ahead with adding project permission, then it means we'll need to add 4 additional permissions:
- view all projects
- view own projects
- view phantom all projects
- view phantom own projects

It is going to be fun to handle all different possibilities of permissions on that module ;). I feel this is more of a major refactor that is needed here.

Reply:
Most modules only have site permissions as of now. Since permissions are implemented module-by-module and project permissions are new(er), it's kind of a mixed bag. In this case specifically, a user cannot view any images for a project they are not involved in (and therefore can't really access them), but if they modify the URL directly, they can access those images. So, its more of a security issue. 

I currently have 2 other PRs (#6624, #6640) dealing with project permissions. @cmadjar let me know if you propose closing this PR in this case.

Reply:
@AlexandraLivadas Sorry, I misunderstood the goal of the PR. I think it is fine for now to restrict access to users belonging to the project only for the subpage.

Sorry for the confusion...